Fix Logcollector IT: `test_command_execution_dbg`

### Description
The logcollector integration tests fail for the 4.4.0 branch. The affected test is `test_command_execution_dbg` for the never-ending command case.

```
FAILED test_logcollector/test_command_monitoring/test_command_execution_dbg.py::test_command_execution_dbg[command_does not end]
```

This test was fixed in https://github.com/wazuh/wazuh-qa/pull/2319 development, but some changes in the framework have produced this error appear again.

**Build**:  https://devel.ci.wazuh.info/job/Test_integration/1277/

--- a/tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py
+++ b/tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py
@@ -61,7 +61,7 @@ import pytest
 import sys
 
 from subprocess import check_output
-from wazuh_testing.tools import monitoring, LOG_FILE_PATH
+from wazuh_testing.tools import monitoring
 from wazuh_testing import global_parameters
 import wazuh_testing.logcollector as logcollector
 from wazuh_testing.tools.configuration import load_wazuh_configurations
@@ -80,20 +80,13 @@ local_internal_options = {
     'logcollector.max_lines': '100',
     'logcollector.debug': '2'
 }
-file_structure = [
-    {
-        'folder_path': os.path.join(tempfile.gettempdir(), 'wazuh-testing'),
-        'filename': ['not_ending_command_file.log'],
-    }
-]
 
-not_ending_command_file_full_path = os.path.join(file_structure[0]['folder_path'], file_structure[0]['filename'][0])
 
 parameters = [
     {'LOG_FORMAT': 'command', 'COMMAND': 'echo', 'ALIAS': ''},
     {'LOG_FORMAT': 'command', 'COMMAND': 'echo hello world', 'ALIAS': 'goodbye'},
     {'LOG_FORMAT': 'command', 'COMMAND': 'not_found_command -o option -v', 'ALIAS': ''},
-    {'LOG_FORMAT': 'command', 'COMMAND': f'tail -f {not_ending_command_file_full_path}', 'ALIAS': ''},
+    {'LOG_FORMAT': 'command', 'COMMAND': 'for ((i=0;;i++)); do echo "Line ${i}"; done', 'ALIAS': ''},
     {'LOG_FORMAT': 'command', 'COMMAND': 'ls -R /tmp', 'ALIAS': ''},
     {'LOG_FORMAT': 'command', 'COMMAND': 'cat doesntexists.txt', 'ALIAS': ''},
     {'LOG_FORMAT': 'command', 'COMMAND': 'cat "ñ", "テスト", "ИСПЫТАНИЕ", "测试", "اختبار".txt', 'ALIAS': ''},
@@ -102,7 +95,7 @@ parameters = [
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo', 'ALIAS': ''},
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'echo hello world', 'ALIAS': 'goodbye'},
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'not_found_command -o option -v', 'ALIAS': ''},
-    {'LOG_FORMAT': 'full_command', 'COMMAND': f'tail -f {not_ending_command_file_full_path}', 'ALIAS': ''},
+    {'LOG_FORMAT': 'full_command', 'COMMAND': 'for ((i=0;;i++)); do echo "Line ${i}"; done', 'ALIAS': ''},
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'ls -R /tmp', 'ALIAS': ''},
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'cat doesntexists.txt', 'ALIAS': ''},
     {'LOG_FORMAT': 'full_command', 'COMMAND': 'cat "ñ", "テスト", "ИСПЫТАНИЕ", "测试", "اختبار".txt', 'ALIAS': ''},
@@ -113,8 +106,8 @@ metadata = [
     {'log_format': 'command', 'command': 'echo', 'alias': '', 'info': 'empty_output'},
     {'log_format': 'command', 'command': 'echo hello world', 'alias': 'goodbye', 'info': 'check_output_and_alias'},
     {'log_format': 'command', 'command': 'not_found_command -o option -v', 'alias': '', 'info': 'not_found'},
-    {'log_format': 'command', 'command': f'tail -f {not_ending_command_file_full_path}', 'alias': '',
-     'info': 'does not end'},
+    {'log_format': 'command', 'command': 'for ((i=0;;i++)); do echo "Line ${i}"; done', 'alias': '',
+     'info': 'does_not_end'},
     {'log_format': 'command', 'command': 'ls -R /tmp', 'alias': '', 'info': 'long_output'},
     {'log_format': 'command', 'command': 'cat doesntexists.txt', 'alias': '', 'info': 'that_fails'},
     {'log_format': 'command', 'command': 'cat "ñ", "テスト", "ИСПЫТАНИЕ", "测试", "اختبار".txt', 'alias': '',
@@ -125,7 +118,7 @@ metadata = [
     {'log_format': 'full_command', 'command': 'echo', 'alias': '', 'info': 'empty_output'},
     {'log_format': 'full_command', 'command': 'echo hello world', 'alias': 'goodbye', 'info': 'check_output_and_alias'},
     {'log_format': 'full_command', 'command': 'not_found_command -o option -v', 'alias': '', 'info': 'not_found'},
-    {'log_format': 'full_command', 'command': f'tail -f {not_ending_command_file_full_path}', 'alias': '',
+    {'log_format': 'full_command', 'command': 'for ((i=0;;i++)); do echo "Line ${i}"; done', 'alias': '',
      'info': 'does not end'},
     {'log_format': 'full_command', 'command': 'ls -R /tmp', 'alias': '', 'info': 'long_output'},
     {'log_format': 'full_command', 'command': 'cat doesntexists.txt', 'alias': '', 'info': 'that_fails'},
@@ -188,14 +181,8 @@ def dbg_reading_command(command, alias, log_format):
                             error_message=logcollector.GENERIC_CALLBACK_ERROR_COMMAND_MONITORING)
 
 
-
-@pytest.fixture(scope="module")
-def get_files_list():
-    """Get file list to create from the module."""
-    return file_structure
-
-def test_command_execution_dbg(get_files_list, create_file_structure_module, configure_local_internal_options_module,
-                               get_configuration, file_monitoring, configure_environment, restart_logcollector):
+def test_command_execution_dbg(configure_local_internal_options_module, get_configuration, file_monitoring,
+                               configure_environment, restart_logcollector):
     '''
     description: Check if the 'wazuh-logcollector' daemon generates debug logs when running commands with
                  special characteristics. For this purpose, the test will configure the logcollector to run
@@ -258,11 +245,6 @@ def test_command_execution_dbg(get_files_list, create_file_structure_module, con
                       callback=logcollector.callback_running_command(log_format=config['log_format'],
                                                                      command=config['command'],
                                                                      escape=True))
-
-    if config['info'] == 'does not end':
-        with open(not_ending_command_file_full_path, 'a') as command_file:
-            for line in range(int(local_internal_options['logcollector.max_lines']) + 5):
-                command_file.write(f"Testing not ending command.Line {line}\n")
 
     # Command with known output to test "Reading command message: ..."
     if config['command'].startswith('echo') and config['alias'] != '':

--- a/tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py
+++ b/tests/integration/test_logcollector/test_command_monitoring/test_command_execution_dbg.py
@@ -261,7 +261,7 @@ def test_command_execution_dbg(get_files_list, create_file_structure_module, con
 
     if config['info'] == 'does not end':
         with open(not_ending_command_file_full_path, 'a') as command_file:
-            for line in range(int(local_internal_options['logcollector.max_lines']) + 1):
+            for line in range(int(local_internal_options['logcollector.max_lines']) + 5):
                 command_file.write(f"Testing not ending command.Line {line}\n")
 
     # Command with known output to test "Reading command message: ..."

